### PR TITLE
Add minReadySeconds to the documentation

### PR DESCRIPTION
**Pull request by FredrikAtIBM**

### CODE REVIEW

File: bundle/manifests/instana-agent-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -326,6 +332,30 @@ spec:
...
+          - list
+          - watch
+        - apiGroups:
+          - ""
+          resources:
+          - pods/log
+          verbs:
+          - get
+          - list
+          - watch
```
Comment (FelixMarxIBM) [inline bundle/manifests/instana-agent-operator.clusterserviceversion.yaml:358]: **question:** Is there a reason why these have changed?

**Comment by konrad-ohms** [comment]
The reason for these changes @FelixMarxIBM are that in the past people did changes without regenerating the bundle manifest files... :upside_down_face: 
Once any `+kubebuilder` comments are changed, also `make manifests` **and** `make bundle` should be executed to generate the proper yaml files around it, also for the Operator Lifecycle Manager (OLM). As some changes were just whitespace changes, I assume someone changed them manually instead of relying on the operator sdk tooling.
I needed to recreate those yaml files, because I needed to adjust `api/v1/inline_types.go` to accept new properties for `minReadySeconds`. Once I generated files, the other changes were introduced as well.

As the changes did just affect the bundle files and not the clusterrole directly, I assume that we claimed to use less permissions in OCP catalogs than we actually required, not sure if the OLM framework would have enforced the limits somehow, but there was clearly a mismatch.

File: bundle/manifests/instana-agent-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -244,6 +247,9 @@ spec:
         - nonResourceURLs:
           - /version
           - /healthz
+          - /metrics
+          - /stats/summary
+          - /metrics/cadvisor
```
[the remark this answers is not in the document]
Comment (konrad-ohms) [reply bundle/manifests/instana-agent-operator.clusterserviceversion.yaml:252]: Same here, they are not new, they are used on main in the cluster role already, the bundle was not regenerated properly: https://github.com/instana/instana-agent-operator/blob/8f444abf0fe2db67f9c513958f76d43effb97536/config/rbac/role.yaml#L90-L92

File: bundle/manifests/instana-agent-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (konrad-ohms) [reply bundle/manifests/instana-agent-operator.clusterserviceversion.yaml:358]: They are already used on the ClusterRole in 
https://github.com/instana/instana-agent-operator/blob/8f444abf0fe2db67f9c513958f76d43effb97536/config/rbac/role.yaml#L175-L182 and https://github.com/instana/instana-agent-operator/blob/8f444abf0fe2db67f9c513958f76d43effb97536/config/rbac/role.yaml#L191-L198

**Review by konrad-ohms — APPROVED** [review]

**Comment by konrad-ohms** [comment]
I manually tested the changes on a GKE cluster now, to see that the changes introduced by https://github.com/instana/helm-charts/commit/81d40c8ab936620f0271cb9eb0af11b6e532797a are also available inside the operator:

I produced a custom operator image build based on the current branch and pushed it to a image registry.
First I exported variables:
```
export IMG=gcr.io/myregistry/instana-agent-operator:konrad
export AGENT_IMG=icr.io/instana/agent:latest
export BUNDLE_IMG=gcr.io/myregistry/instana-agent-operator-bundle:konrad
```
Afterwards I build the manifests and bundle files via make targets (needed to use an container with go version go1.18.10 linux/amd64, my local version was too recent.

I deployed the operator via `make install` and `make deploy`.
Once the operator was running, I used this instana-agent artifact to trigger a deployment:

```
$ cat operator-v2-minreadyseconds.yaml 
apiVersion: instana.io/v1
kind: InstanaAgent
metadata:
  name: instana-agent
  namespace: instana-agent
spec:
  zone:
   name: konrad-gke # (optional) name of the zone of the host
  cluster:
   name: konrad-gke
  agent:
   key: xxx
   endpointHost: xxx.instana.io
   endpointPort: "443"
   # The minimum number of seconds for which a newly created Pod should be ready without any of its containers crashing, for it to be considered available
   minReadySeconds: 15
   env: {}
   configuration_yaml: |
    # You can leave this empty, or use this to configure your instana agent.
    # See https://docs.instana.io/setup_and_manage/host_agent/on/kubernetes/
  k8s_sensor:
    deployment:
      # Specifies whether or not to enable the Deployment and turn off the Kubernetes sensor in the DaemonSet
      enabled: true
      # Use three replicas to ensure the HA by the default.
      replicas: 3
      # The minimum number of seconds for which a newly created Pod should be ready without any of its containers crashing, for it to be considered available
      minReadySeconds: 16
  kubernetes:
    # Configures use of a Deployment for the Kubernetes sensor rather than as a potential member of the DaemonSet. Is only accepted if k8s_sensor.deployment.enabled=false
    deployment:
      # Specifies whether or not to enable the Deployment and turn off the Kubernetes sensor in the DaemonSet
      enabled: false
      # Use a single replica, the impact will generally be low and we need to address a host of other concerns where clusters are large.
      replicas: 1
      # The minimum number of seconds for which a newly created Pod should be ready without any of its containers crashing, for it to be considered available
      minReadySeconds: 17
```

I checked the expected properties afterwards:
```
$ kubectl get all -n instana-agent
NAME                                     READY   STATUS    RESTARTS   AGE
pod/controller-manager-d7496dc85-hg4hj   1/1     Running   0          9m34s
pod/controller-manager-d7496dc85-zcz2w   1/1     Running   0          9m34s
pod/instana-agent-9zgdl                  1/1     Running   0          8m20s
pod/instana-agent-x4tbm                  1/1     Running   0          8m20s
pod/instana-agent-xwwwd                  1/1     Running   0          8m20s
pod/k8sensor-666d8db8cd-nzp8k            1/1     Running   0          8m20s
pod/k8sensor-666d8db8cd-xc5xx            1/1     Running   0          8m20s
pod/k8sensor-666d8db8cd-zbmlw            1/1     Running   0          8m20s

NAME                             TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)                                 AGE
service/instana-agent            ClusterIP   10.60.3.53   <none>        42699/TCP,55680/TCP,4317/TCP,4318/TCP   8m21s
service/instana-agent-headless   ClusterIP   None         <none>        42699/TCP,55680/TCP,4317/TCP,4318/TCP   8m21s

NAME                           DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/instana-agent   3         3         3       3            3           <none>          8m20s

NAME                                 READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/controller-manager   2/2     2            2           9m35s
deployment.apps/k8sensor             3/3     3            3           8m21s

NAME                                           DESIRED   CURRENT   READY   AGE
replicaset.apps/controller-manager-d7496dc85   2         2         2       9m35s
replicaset.apps/k8sensor-666d8db8cd            3         3         3       8m21s
```

Afterwards I checked if the templates from helm were correctly populated so that the agent deamonset used the right minReadySeconds value:

``` 
$ kubectl get -n instana-agent daemonset.apps/instana-agent -o yaml | grep -B 26 minReadySeconds
kind: DaemonSet
metadata:
  annotations:
    deprecated.daemonset.template.generation: "1"
    meta.helm.sh/release-name: instana-agent
    meta.helm.sh/release-namespace: instana-agent
  creationTimestamp: "2024-04-19T09:42:54Z"
  generation: 1
  labels:
    app.kubernetes.io/instance: instana-agent
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: instana-agent
    app.kubernetes.io/version: 1.2.72
    helm.sh/chart: instana-agent-1.2.72
  name: instana-agent
  namespace: instana-agent
  ownerReferences:
  - apiVersion: instana.io/v1
    blockOwnerDeletion: true
    controller: true
    kind: InstanaAgent
    name: instana-agent
    uid: 334b76b1-47f1-411b-b690-03e68de8f10b
  resourceVersion: "344566845"
  uid: b6b94242-38fb-4ce9-a626-96c7023d3d1a
spec:
  minReadySeconds: 15
```

The k8sensor looks good as well:

```
$ kubectl get -n instana-agent deployment.apps/k8sensor -o yaml | grep -B 27 minReadySeconds
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
    meta.helm.sh/release-name: instana-agent
    meta.helm.sh/release-namespace: instana-agent
  creationTimestamp: "2024-04-19T09:42:54Z"
  generation: 1
  labels:
    app: k8sensor
    app.kubernetes.io/instance: instana-agent
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: instana-agent
    app.kubernetes.io/version: 1.2.72
    helm.sh/chart: instana-agent-1.2.72
  name: k8sensor
  namespace: instana-agent
  ownerReferences:
  - apiVersion: instana.io/v1
    blockOwnerDeletion: true
    controller: true
    kind: InstanaAgent
    name: instana-agent
    uid: 334b76b1-47f1-411b-b690-03e68de8f10b
  resourceVersion: "344566653"
  uid: 957624d1-5a54-4089-9901-9f561aa8f34b
spec:
  minReadySeconds: 16
```

**Pull request merged**